comictaggerlib/ui/talkeruigenerator.py

```diff
@@ -64,6 +73,12 @@ def call_check_api(*args: Any, le_url: QtWidgets.QLineEdit, le_key: QtWidgets.QL
         # To enable setting and getting
         sources["tabs"][talker_id].widgets[f"talker_{talker_id}_{talker_id}_url"] = le_url
 
+    # The key button row was recorded so we add the show/hide button
+    if btn_show_key is not None:
+        btn_show = QtWidgets.QPushButton("Show/Hide")
+        layout.addWidget(btn_show, btn_show_key, 2)
+        btn_show.clicked.connect(partial(show_key, le_key=le_key))
+
     # The button row was recorded so we add it
     if btn_test_row is not None:
```

Comment:
You renamed btn_test_row and added a new check but left the old one in?

Reply:
No. It's a new button to show and hide the `key` field which is now a password by default.

![image](https://github.com/comictagger/comictagger/assets/1141189/5705708d-5098-4863-a34b-678f795b91c2)

Reply:
This now no longer has a `Test API` button if the url setting is disabled `talker_url.file == False` and the key setting is enabled `talker_key.file == True` where as before it still had the button. This will also create an exception if `talker_key.file == False`

Reply:
Okay, yes. I had forgotten we'd talked about this before and how the test button works. I think the best solution will then be this way: https://kushaldas.in/posts/creating-password-input-widget-in-pyqt.html

#442 is to address the problem of a missing key or url.

Reply:
Yes that will be better

Reply:
I used an 80% grey icon to try and cover light and dark. I used the icons from https://github.com/pythonguis/python-qtwidgets which should be fine?

Reply:
Yes, they use the MIT License which is fine.